course/src/main/java/org/openedx/course/presentation/calendarsync/CalendarManager.kt

```diff
@@ -0,0 +1,396 @@
+package org.openedx.course.presentation.calendarsync
+
+import android.annotation.SuppressLint
+import android.content.ContentUris
+import android.content.ContentValues
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.database.Cursor
+import android.net.Uri
+import android.provider.CalendarContract
+import androidx.core.content.ContextCompat
+import org.openedx.core.data.storage.CorePreferences
+import org.openedx.core.domain.model.CourseDateBlock
+import org.openedx.core.system.ResourceManager
+import org.openedx.core.utils.Logger
+import org.openedx.core.utils.toCalendar
+import org.openedx.course.R
+import java.util.Calendar
+import java.util.TimeZone
+import java.util.concurrent.TimeUnit
+import org.openedx.core.R as CoreR
+
+class CalendarManager(
+    private val context: Context,
+    private val corePreferences: CorePreferences,
+    private val resourceManager: ResourceManager,
+) {
+    private val logger = Logger(CalendarManager::class.java.name)
```

Comment:
CalendarManager::class.java.name can be changed during obfuscation, it's better to use contants here